# 2020-09-10 (木曜日) 

## Django Channels: `django.core.exceptions.SynchronousOnlyOperation: You cannot call this from an async context - use a thread or sync_to_async.`

- [SynchronousOnlyOperation Error in with django 3 and django channels](https://stackoverflow.com/questions/59632125/synchronousonlyoperation-error-in-with-django-3-and-django-channels)
- [database_sync_to_async](https://channels.readthedocs.io/en/latest/topics/databases.html?highlight=database_sync_to_async#database-sync-to-async)

## Django Channels: AsyncJsonWebsocketConsumer

- [Python AsyncJsonWebSocketConsumer Problem](https://stackoverflow.com/questions/59304466/python-asyncjsonwebsocketconsumer-problem)
- [JsonWebsocketConsumer](https://channels.readthedocs.io/en/latest/topics/consumers.html#jsonwebsocketconsumer)

## Django: マイグレーション: `django.db.utils.OperationalError: (1091, "Can't DROP 'unique_recipient'; check that column/key exists")`

- https://stackoverflow.com/questions/15793990/django-how-to-set-foreign-key-checks-to-0
- https://stackoverflow.com/questions/8051556/django-how-to-prevent-database-foreign-key-constraint-creation
- [ForeignKey.db_constraint](https://docs.djangoproject.com/en/dev/ref/models/fields/#django.db.models.ForeignKey.db_constraint)


## Django: DRFが返すURLのスキームが `http` になってしまう問題

- [request.is_secure() always returns false with uwsgi server](https://stackoverflow.com/questions/11441832/request-is-secure-always-returns-false-with-uwsgi-server)


[settings.py](https://docs.djangoproject.com/en/3.1/ref/settings/#secure-proxy-ssl-header):

~~~py
# Set a tuple with two elements – the name of the header to look for and the required value.
SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
~~~